### PR TITLE
Trim all duplicate slashes

### PR DESCRIPTION
**Pull request by kidonng**
Though very rare, it is possible to have multiple ~~leading and trailing~~ 
successive slashes in `pathname`.

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -642,7 +645,7 @@ collect.set('isNewRepoTemplate', [
 const getUsername = (): string | undefined => document.querySelector('meta[name="user-login"]')?.getAttribute('content')!;
 
 /** Drop leading and trailing slashes */
```
Comment (yakov116) [inline index.ts:647]: update the comment?

Comment (kidonng) [reply index.ts:647]: Hmm I think it's fine? Not _super_ clear though

File: index.ts [hunk 2]
```diff
@@ -56,10 +56,13 @@ collect.set('isCompare', [
 
 export const isDashboard = (url: URL | HTMLAnchorElement | Location = location): boolean => !isGist(url) && /^$|^(orgs\/[^/]+\/)?dashboard(\/|$)/.test(getCleanPathname(url));
 collect.set('isDashboard', [
+	'https://github.com///',
```
Comment (kidonng) [inline index.ts:59]: As it stands two trailing slashes here is OK without this patch so an extra test case with three is needed.

File: index.ts [hunk 3]
```diff
@@ -641,8 +644,8 @@ collect.set('isNewRepoTemplate', [
 /** Get the logged-in user’s username */
 const getUsername = (): string | undefined => document.querySelector('meta[name="user-login"]')?.getAttribute('content')!;
 
-/** Drop leading and trailing slashes */
-const getCleanPathname = (url: URL | HTMLAnchorElement | Location = location): string => url.pathname.slice(1, url.pathname.endsWith('/') ? -1 : undefined);
+/** Drop all leading and trailing slashes */
+const getCleanPathname = (url: URL | HTMLAnchorElement | Location = location): string => url.pathname.replace(/^\/+(.*?)\/*$/, '$1');
```
Comment (fregante) [inline index.ts:648]: This covers slashes in the middle too

```suggestion
const getCleanPathname = (url: URL | HTMLAnchorElement | Location = location): string => url.pathname.replaceAll('//', '/').slice(1, url.pathname.endsWith('/') ? -1 : undefined);
Write
```

Comment (kidonng) [reply index.ts:648]: Sometimes the slashes in the middle do make a difference though (leading to the 403 page), however I can't find an example like I used to see now.

Comment (fregante) [reply index.ts:648]: > slashes in the middle do make a difference

I doubt that's intended/acceptable, URLs should not have multiple slashes

Comment (kidonng) [reply index.ts:648]: > > slashes in the middle do make a difference
> 
> I doubt that's intended/acceptable, URLs should not have multiple slashes

In case I didn't make it clear, I was only referring to GitHub, not in a general sense. That is indeed, as you described, an _acceptable_ (or let's say, fault-tolerant) behavior. However we are not GitHub, we can't ensure that multiple slashes in the middle does not matter.

Comment (fregante) [reply index.ts:648]: This PR is about dropping duplicate slashes.

Drop all duplicate slashes or close the PR. Don't claim that duplicate leading/trailing slashes are not important and then argue that duplicate slashes in the middle are important. It's either both or neither.

Comment (kidonng) [reply index.ts]: > Don't claim that duplicate leading/trailing slashes are not meaningful and then argue that duplicate slashes in the middle are.

I didn't describe them as "meaningful", I was just pointing out neither way covers all cases. Skipping slashes in the middle ensure the pathname before and after processing return the same page, which is better in my opinion.

I do understand your intentions though, as I said I struggle to find the page I used to notice things are different, so those edge cases are rare enough to be safely ignored.

**Comment by fregante** [comment]
By the way the issue was that the tests are running on a node version that doesn’t have replaceAll. 

But the current solution is ok.
